File: php/features.php
<?php
declare(strict_types=1);

// Fortnite SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class FortniteFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new FortniteBaseFeature();
            case "test":
                return new FortniteTestFeature();
            default:
                return new FortniteBaseFeature();
        }
    }
}

File: py/features.py
# Fortnite SDK feature factory

from feature.base_feature import FortniteBaseFeature
from feature.test_feature import FortniteTestFeature


def _make_feature(name):
    features = {
        "base": lambda: FortniteBaseFeature(),
        "test": lambda: FortniteTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
